internal/controllers/preflightvalidation_reconciler.go

```diff
@@ -0,0 +1,249 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/api/v1beta2"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/api"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/filter"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/metrics"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/mic"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/module"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/preflight"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	PreflightValidationReconcilerName = "PreflightValidation"
+)
+
+// PreflightReconciler reconciles a PreflightValidation object
+type preflightValidationReconciler struct {
+	client       client.Client
+	filterAPI    *filter.Filter
+	metricsAPI   metrics.Metrics
+	helper       preflightReconcilerHelper
+	preflightAPI preflight.PreflightAPI
+}
+
+func NewPreflightValidationReconciler(
+	client client.Client,
+	filterAPI *filter.Filter,
+	metricsAPI metrics.Metrics,
+	micAPI mic.MIC,
+	kernelAPI module.KernelMapper,
+	preflightAPI preflight.PreflightAPI,
+) *preflightValidationReconciler {
+	helper := newPreflightReconcilerHelper(client, micAPI, metricsAPI, kernelAPI, preflightAPI)
+	return &preflightValidationReconciler{
+		client:       client,
+		filterAPI:    filterAPI,
+		metricsAPI:   metricsAPI,
+		helper:       helper,
+		preflightAPI: preflightAPI}
+}
+
+func (r *preflightValidationReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(PreflightValidationReconcilerName).
+		For(&v1beta2.PreflightValidation{}, builder.WithPredicates(filter.PreflightReconcilerUpdatePredicate())).
+		Owns(&kmmv1beta1.ModuleImagesConfig{}).
+		Watches(
+			&kmmv1beta1.Module{},
+			handler.EnqueueRequestsFromMapFunc(r.filterAPI.EnqueueAllPreflightValidations),
+			builder.WithPredicates(filter.PreflightReconcilerUpdatePredicate()),
+		).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: 1,
+		}).
+		Complete(
+			reconcile.AsReconciler[*v1beta2.PreflightValidation](r.client, r),
+		)
+}
+
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=modules,verbs=get;list;watch
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=preflightvalidations,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=preflightvalidations/status,verbs=get;update;patch
```

Comment:
Let's move those RBAC annotation to the `Module` controller.

@TomerNewman recently gathered them together to address the spoke issue not getting the correct RBAC: https://github.com/kubernetes-sigs/kernel-module-management/pull/1102.

If we plan to also support pre-flight in the hub topology then we can copy them to the `ManagedClusterModule` controller as well.